zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/Publisher.java

```diff
@@ -117,61 +129,143 @@ public static boolean Publish(Collection collection, String email, CollectionRea
+
+        boolean published = false;
+
+        boolean filesPublished = publishFiles(
+                collection,
+                collectionReader,
+                encryptionPassword,
+                userEmail);
+
+        if (filesPublished) {
+            published = publishDatasets(collection);
+        }
+
+        return published;
+    }
+
+    private boolean publishDatasets(Collection collection) throws IOException {
+
+        boolean datasetsPublished = false;
+
+        try {
+            logDebug("publishing api datasets for collection").collectionName(collection).log();
+            datasetService.publishDatasetsInCollection(collection);
+            collection.description.publishEndDate = new Date();
+            datasetsPublished = true;
+
+        } catch (Exception e) {
+            SlackNotification.alarm(String.format("Exception setting API dataset to published : %s: %s", collection.description.name, e.getMessage()));
+            logError(e, "exception publishing datasets").collectionName(collection).log();
+
+        } finally {
+            // Save any updates to the collection
+            collection.save();
+        }
+
+        return datasetsPublished;
+    }
+
+    /**
+     * aggregate the pre-publish and publish steps into one function.
      *
      * @param collection The collection to be published.
      * @param email      An identifier for the publishing user.
      * @return If publishing succeeded, true.
      * @throws IOException If a general error occurs.
      */
-    public static boolean PublishFilesToWebsite(Collection collection, String email, CollectionReader collectionReader) throws IOException {
+    private boolean FullPublish(Collection collection, String email, CollectionReader collectionReader) {
 
         boolean publishComplete = false;
         String encryptionPassword = Random.password(100);
         try {
-            collection.description.publishStartDate = new Date();
-            BeginPublish(collection, encryptionPassword);
-            SendManifest(collection, encryptionPassword);
-            PublishFilteredCollectionFiles(collection, collectionReader, encryptionPassword);
-            publishComplete = CommitPublish(collection, email, encryptionPassword);
-            collection.description.publishEndDate = new Date();
+
+            DoPrePublish(collection, encryptionPassword);
+            publishComplete = DoPublish(collection, collectionReader, encryptionPassword, email);
 
         } catch (IOException e) {
+            logError(e, "Exception publishing collection").collectionName(collection).log();
+        }
+
+        return publishComplete;
+    }
+
+    private static boolean publishFiles(Collection collection, CollectionReader collectionReader, String encryptionPassword, String userEmail) throws IOException {
+
+        boolean filesPublished = false;
+
+        try {
+            collection.description.publishStartDate = new Date();
```

Comment:
can you use `collection.getDescription()` if it exists

Reply:
replace all cases in this file